src/sonic-config-engine/portconfig.py

```diff
@@ -172,85 +172,152 @@ def parse_port_config_file(port_config_file):
+            self._parse_supported_speed(supported_speed)
+            self.num_assigned_lanes = self._parse_num_assigned_lanes(num_assigned_lanes)
+
+        @classmethod
+        def _speed_to_int(cls, speed):
+            try:
+                if speed.endswith('G'):
+                    return int(speed.replace('G', '')) * 1000
+
+                return int(speed)
+            except ValueError:
+                raise RuntimeError("Unsupported speed format '{}'".format(speed))
+
+        def _parse_supported_speed(self, speed):
+            if not speed:
+                return
+
+            if not speed.startswith('[') and not speed.endswith(']'):
+                raise RuntimeError("Unsupported port breakout format!")
+
+            for s in speed[1:-1].split(','):
+                self.supported_speed.add(self._speed_to_int(s.strip()))
+
+        def _parse_num_assigned_lanes(self, num_assigned_lanes):
+            if not num_assigned_lanes:
+                return
+
+            if isinstance(num_assigned_lanes, int):
+                return num_assigned_lanes
+
+            if not num_assigned_lanes.startswith('(') and not num_assigned_lanes.endswith(')'):
+                raise RuntimeError("Unsupported port breakout format!")
+
+            return int(num_assigned_lanes[1:-1])
+
+        def __eq__(self, other):
+            if isinstance(other, BreakoutCfg.BreakoutModeEntry):
+                if self.num_ports != other.num_ports:
+                    return False
+                if self.supported_speed != other.supported_speed:
+                    return False
+                if self.num_assigned_lanes != other.num_assigned_lanes:
+                    return False
+                return True
             else:
-                raise Exception('Regex return for speed is None...')
+                return False
 
-            ports[intf_name]['index'] = index.split(",")[alias_position]
+        def __ne__(self, other):
+            return not self == other
 
-            parent_intf_id += step
-            alias_position += 1
-            lanes_start += step
+        def __hash__(self):
+            return hash((self.num_ports, tuple(self.supported_speed), self.num_assigned_lanes))
+
+    def __init__(self, name, bmode, properties):
+        self._interface_base_id = int(name.replace(PORT_STR, ''))
+        self._properties = properties
+        self._lanes = properties ['lanes'].split(',')
+        self._indexes = properties ['index'].split(',')
+        self._breakout_mode_entry = self._str_to_entries(bmode)
+        self._breakout_capabilities = None
+
+        # Find specified breakout mode in port breakout mode capabilities
+        for supported_mode in self._properties['breakout_modes']:
+            if self._breakout_mode_entry == self._str_to_entries(supported_mode):
+                self._breakout_capabilities = self._properties['breakout_modes'][supported_mode]
+                break
```

Comment:
why did you use break here? 
what will be case when `self._breakout_mode_entry != self._str_to_entries(supported_mode)`? Should we address such cases?

Reply:
Here we are trying to find whether the given breakout mode is supported. We are iterating over supported modes and checking if the given breakout mode is equal to one of them. If the breakout mode is not equal to any of the supported modes we will raise an exception with an appropriated message at line 250.

Reply:
LGTM!